test/framework/functional/output_cloudwatch.go

```diff
@@ -2,13 +2,99 @@ package functional
 
 import (
 	"context"
+	"crypto/tls"
 	"fmt"
+	"net/http"
 
 	log "github.com/ViaQ/logerr/v2/log/static"
+	"github.com/aws/aws-sdk-go-v2/aws"
 	cwl "github.com/aws/aws-sdk-go-v2/service/cloudwatchlogs"
+	openshiftv1 "github.com/openshift/api/route/v1"
+	logging "github.com/openshift/cluster-logging-operator/api/logging/v1"
+	"github.com/openshift/cluster-logging-operator/internal/runtime"
+	v1 "k8s.io/api/core/v1"
 	"k8s.io/apimachinery/pkg/util/wait"
 )
 
+const (
+	AwsAccessKeyID      = "AKIAIOSFODNN7EXAMPLE"
+	AwsSecretAccessKey  = "wJalrXUtnFEMI/K7MDENG/bPxRfiCYEXAMPLEKEY" //nolint:gosec
+	cloudwatchMotoImage = "quay.io/openshift-logging/moto:2.2.3.dev0"
+	CloudwatchSecret    = "cloudwatch-secret"
+)
+
+var (
+	cwlClient *cwl.Client
+	service   *v1.Service
+	route     *openshiftv1.Route
+)
+
+func (f *CollectorFunctionalFramework) AddCloudWatchOutput(b *runtime.PodBuilder, output logging.OutputSpec) error {
+	if err := f.createCloudWatchService(); err != nil {
```

Comment:
This is a candidate for refactoring even more since @vparfonov  had similar changes for azure.  Functional tests run in the same container and have no need for routes or  services.  For Azure I think we did some hackery to modify the host file.  I don't see any reason the mock CW should run otherwise.